geopyspark/geotrellis/catalog.py

```diff
@@ -475,19 +286,39 @@ class AttributeStore(object):
         hist = store.layer("us-nlcd2011-30m-epsg3857", zoom=7).read("histogram")
         hist = gps.Histogram.from_dict(hist)
     """
-
     def __init__(self, uri):
         self.uri = uri
-        self.underlying = scala_companion("geotrellis.spark.io.AttributeStore").apply(uri)
+        pysc = get_spark_context()
+        try:
+            self.wrapper = pysc._gateway.jvm.geopyspark.geotrellis.io.AttributeStoreWrapper(uri)
+        except Py4JJavaError as err:
+            raise ValueError(err.java_exception.getMessage())
+
+
+    @classmethod
+    def build(cls, uri):
```

Comment:
Its the same deal as `ColorMap.build`, the `__init__` can assign a attributes to a class but it can't choose to return a different instance. For that you have to either use `__new__` or a class method like `build`. The last refactor removed the use of it though (it was confusing what the way it was being used) so it can go away until it is needed.